graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin2/Kotlin2TypeLookup.kt

```diff
@@ -244,6 +244,19 @@ class Kotlin2TypeLookup(
         return NodeTraverser().postOrder(visitor, fieldType) as KtTypeName
     }
 
+    /**
+     * Takes a GQL interface type name and returns the appropriate kotlin type given all of the mappings defined in the schema and config
+     */
```

Comment:
Looks great, if possible we try to maintain feature parity between java and kotlin as much as possible. Would you be able to add that or make a note that this is not supported for Java?

Reply:
Oh yeah, totally forgot about the Java side. I'll add that tomorrow

Reply:
@srinivasankavitha I pushed a commit adding this to the other kotlin and java generators. They don't have the same test suite so I was unsure how to test it though.

Reply:
Here's an example of a test to verify typemapping: https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt#L1007

Reply:
Thanks for updating java and kotlin generators as well.

Reply:
Actually I just see this for kotlin : https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/KotlinCodeGenTest.kt#L1075
So I'm wondering why this wasn't working for you 🤔

Reply:
oh interesting! I'm not sure. I'll look into that test and make the other updates you commented. Thanks for the feedback and direction on tests!

Reply:
@srinivasankavitha ah I see the issue with that test:

Currently, the test includes mapping both for the interface **and the type implementing it**:

```kotlin
val result = CodeGen(
            CodeGenConfig(
                schemas = setOf(schema),
                packageName = basePackageName,
                language = Language.KOTLIN,
                typeMapping = mapOf(
                    "SomethingWithAName" to "mypackage.SomethingWithAName",
                    "Person" to "mypackage.Person"
                )
            )
        ).generate()
```

DGS does generate the package name for the return type value on the type which returns an interface:
```
type SearchResult {
  item: SomethingWithAName
}
```

This works because of the config type mapping used when finding the correct return type. However, if you remove the mapping of `"Person" to "mypackage.Person"`, without these changes it generates the data class with no reference to the mapped package for the interface and fails compilation.

Reply:
I see, ok, that makes sense.